R/navbar.R

```diff
@@ -50,8 +50,8 @@ navbar_links <- function(navbar, components, depth = 0L, bs_version = 3L) {
   components[names(components_meta)] <- components_meta
   components <- purrr::compact(components)
 
-  right_comp <- intersect(structure$right, names(components))
-  left_comp <- intersect(structure$left, names(components))
+  right_comp <- strict_intersect(structure$right, names(components))
```

Comment:
I think I'd prefer to fix this by coercing `structure$right` to a character vector. Maybe something like this:

```R
yaml_character <- function(x, path) {
  if (identical(x, list())) {
    character()
  } else if (is.character(x)) {
    x
  } else {
    "Some error that `path` needs to be empty or a character vector"
  }
}
```

That has the added benefit of giving a clear error if you (e.g.) put a number there.